### ISSUE TITLE
Compose module error

### ISSUE THREAD

**Issue by michK** [GiovineItalia/Gadfly.jl#971]
When trying to do `using Gadfly` I get the error:

```
INFO: Precompiling module Gadfly.
ERROR: LoadError: ArgumentError: Module Compose not found in current path.
Run `Pkg.add("Compose")` to install the Compose package.
 in require(::Symbol) at ./loading.jl:365
 in include_from_node1(::String) at ./loading.jl:488
 in macro expansion; at ./none:2 [inlined]
 in anonymous at ./<missing>:?
 in eval(::Module, ::Any) at ./boot.jl:234
 in process_options(::Base.JLOptions) at ./client.jl:242
 in _start() at ./client.jl:321
while loading /home/user/.julia/v0.5/Gadfly/src/Gadfly.jl, in expression starting on line 7
ERROR: Failed to precompile Gadfly to /home/mkruger/.julia/lib/v0.5/Gadfly.ji.
 in compilecache(::String) at ./loading.jl:593
 in require(::Symbol) at ./loading.jl:422
```

Both the `Gadfly` and `Compose` packages have however been 'successfully' installed, i.e I did Pkg.add() for both and they seem to have installed, but I cannot import either of them.

**Comment by tlnagy** [comment]
What version of Julia are you on? Also, try doing `Pkg.build("Compose")` and `Pkg.build("Gadfly")` and see if that helps.

**Comment by michK** [comment]
I am running Julia 0.5.1.

I tried your suggestions, but still get the same problem.

I updated the question to show the entire REPL output when I do `using Gadfly`, perhaps that helps.

Looking [here](https://github.com/JuliaPlots/Plots.jl/issues/328), it seems like it might have something to do with some package version requirement, but not exactly sure.

**Comment by tlnagy** [comment]
What's the output of `Pkg.status()`?

**Comment by michK** [comment]
```
11 required packages:
 - Cairo                         0.3.0
 - Compose                       0.4.5              2337062a (dirty)
 - DataFrames                    0.9.0
 - Fontconfig                    0.1.1
 - Gadfly                        0.6.0
 - IJulia                        1.4.1
 - Primes                        0.1.3
 - PyCall                        1.11.1
 - PyPlot                        2.3.1
 - RDatasets                     0.2.0
 - ScikitLearn                   0.3.0
54 additional packages:
 - AxisAlgorithms                0.1.6
 - BinDeps                       0.4.7
 - Calculus                      0.2.2
 - ColorTypes                    0.4.0
 - Colors                        0.7.3
 - Compat                        0.22.0
 - Conda                         0.5.3
 - Contour                       0.2.0
 - DataArrays                    0.3.12
 - DataStructures                0.5.3
 - DiffBase                      0.1.0
 - Distances                     0.4.1
 - Distributions                 0.12.2
 - DualNumbers                   0.3.0
 - FileIO                        0.3.1
 - FixedPointNumbers             0.3.6
 - FixedSizeArrays               0.2.5
 - ForwardDiff                   0.4.1
 - GZip                          0.3.0
 - Graphics                      0.2.0
 - Hexagons                      0.0.4
 - Hiccup                        0.1.1
 - Interpolations                0.3.8
 - Iterators                     0.3.0
 - JSON                          0.8.3
 - Juno                          0.2.7
 - KernelDensity                 0.3.2
 - LaTeXStrings                  0.2.1
 - LineSearches                  0.1.5
 - Loess                         0.1.0
 - MacroTools                    0.3.6
 - Measures                      0.0.3
 - Media                         0.2.6
 - NaNMath                       0.2.4
 - Nettle                        0.3.0
 - Optim                         0.7.8
 - PDMats                        0.5.6
 - Parameters                    0.7.2
 - PositiveFactorizations        0.0.4
 - QuadGK                        0.1.2
 - RData                         0.0.4
 - Ratios                        0.0.4
 - Reexport                      0.0.3
 - Rmath                         0.1.6
 - SHA                           0.3.2
 - ScikitLearnBase               0.3.0
 - Showoff                       0.0.7
 - SortingAlgorithms             0.1.1
 - SpecialFunctions              0.1.1
 - StatsBase                     0.13.1
 - StatsFuns                     0.4.0
 - URIParser                     0.1.8
 - WoodburyMatrices              0.2.2
 - ZMQ   
```

**Comment by michK** [comment]
I found the problem.

As discussed [here](https://groups.google.com/forum/#!topic/julia-users/SmyOHSfgqQg) the package, in my case `Compose` became 'dirty' because I had made a change manually in that packages directory, which is a Git directory, and thus Julia's package manager can't touch the package.

To solve the problem I did `git checkout .` in the `/home/user/.julia/v0.5/Compose` directory, since the change I made in that directory was not intentional. This solved my problem.

Note that `git checkout .` will delete all your changes made in that package directory, so use the appropriate Git workflow to sort out what you did in the package directory that is causing the package to be 'dirty' if you made some intentional changes there.
